## 데이터 취합_라벨링 파일 만들기_데이터 랜덤 샘플링
- ref. https://zephyrus1111.tistory.com/49
- 
- 코더1팀(차미/상훈/정은); 아프니까 사장이다, 브런치
- 코더2팀(혜진/연수/채란); 셀러오션
- 코더3팀(유빈/새연/다운); 지식인

In [1]:
from pandas import DataFrame
import pandas as pd
import numpy as np
import glob
import os

### 브런치 파일 통합

In [2]:
brunch_list = glob.glob('*브런치*.csv')

In [3]:
dfs = []
for file in brunch_list:
    df = pd.read_csv(file)
    print(file)
    dfs.append(df) 

[KB]브런치_상훈_전처리_final.csv
[KB]브런치_정은_전처리_final.csv
[KB]브런치_차미_전처리_final.csv
[KB]브런치_채널통합.csv
브런치_라벨링.csv


In [4]:
df_brunch = pd.concat(dfs, ignore_index = True)
print(df_brunch.shape)
df_brunch.head()

(25093, 5)


Unnamed: 0                                             review        date  \
0        18.0     프리패스  상황에 맞는 OOTD가 이렇게나 중요하다 택배박스도 보여주고 어디 ...  2023-09-15   
1        19.0  틈은 흠일까  지난 시간들을 돌이켜 보면 나는 원래 틈이 많은 사람이었다  손해를 ...  2023-09-14   
2        26.0                                       무인모텔     계속   2023-09-13   
3        27.0  무인문구점  그게 되겠어      아이가 생기고 나서는 주말은 온전히 아이를 위한 ...  2023-09-13   
4        91.0                      무인키즈카페 창업일기  편  상가 인테리어 계약     2023-09-04   

  keyword  코더A  
0      무인  NaN  
1      무인  NaN  
2      무인  NaN  
3      무인  NaN  
4      무인  NaN

In [5]:
df_brunch.drop(columns=['Unnamed: 0', 'date', 'keyword'], inplace=True)

In [6]:
df_brunch = df_brunch.drop_duplicates()
print(df_brunch.shape)
df_brunch.head()

(12093, 2)


review  코더A
0     프리패스  상황에 맞는 OOTD가 이렇게나 중요하다 택배박스도 보여주고 어디 ...  NaN
1  틈은 흠일까  지난 시간들을 돌이켜 보면 나는 원래 틈이 많은 사람이었다  손해를 ...  NaN
2                                       무인모텔     계속   NaN
3  무인문구점  그게 되겠어      아이가 생기고 나서는 주말은 온전히 아이를 위한 ...  NaN
4                      무인키즈카페 창업일기  편  상가 인테리어 계약     NaN

### 아싸 파일 통합

In [7]:
asa_list = glob.glob('*아싸*.csv')

In [8]:
dfs = []
for file in asa_list:
    df = pd.read_csv(file)
    print(file)
    dfs.append(df) 

[KB]아싸_라이프_차미_전처리_final.csv
[KB]아싸_상훈_전처리_final.csv
[KB]아싸_정은_전처리_final.csv
[KB]아싸_차미_전처리_final.csv
[KB]아싸_채널통합.csv
아싸_라벨링.csv


In [9]:
df_asa = pd.concat(dfs, ignore_index = True)
print(df_asa.shape)
df_asa.head()

(155473, 5)


Unnamed: 0                                             review  코더A  코더B  \
0           0  사업자대출여러건 있을때 아파트입주시 집단대출 사업자대출이 코로나1차2차5000 소상...  NaN  NaN   
1           1  한가하니 이번에 만든 청귤청으로 에이드 만들어 먹었네요ㅎ 말일이라 그런지 조~~용하...  NaN  NaN   
2           2  걱정했는데 손님이 꽤 있네요 여름휴가 보내고 와서 손님 떨어질까봐 걱정했는데 그래도...  NaN  NaN   
3           3  비~~좀 그만~~ 어제 오후부터 쉬지 않고 내리고 있네요. 안전운전들 하시고 즐거운...  NaN  NaN   
4           4  가족의 도박문제 대처 저희 큰아들이 작년에 도박문제로 가족이 많이 힘들었습니다. 1...  NaN  NaN   

   코더C  
0  NaN  
1  NaN  
2  NaN  
3  NaN  
4  NaN

In [10]:
df_asa.drop(columns=['Unnamed: 0'], inplace=True)

In [11]:
df_asa = df_asa.drop_duplicates()
print(df_asa.shape)
df_asa.head()

(76923, 4)


review  코더A  코더B  코더C
0  사업자대출여러건 있을때 아파트입주시 집단대출 사업자대출이 코로나1차2차5000 소상...  NaN  NaN  NaN
1  한가하니 이번에 만든 청귤청으로 에이드 만들어 먹었네요ㅎ 말일이라 그런지 조~~용하...  NaN  NaN  NaN
2  걱정했는데 손님이 꽤 있네요 여름휴가 보내고 와서 손님 떨어질까봐 걱정했는데 그래도...  NaN  NaN  NaN
3  비~~좀 그만~~ 어제 오후부터 쉬지 않고 내리고 있네요. 안전운전들 하시고 즐거운...  NaN  NaN  NaN
4  가족의 도박문제 대처 저희 큰아들이 작년에 도박문제로 가족이 많이 힘들었습니다. 1...  NaN  NaN  NaN

### 셀러오션 파일 통합

In [12]:
sell_list = glob.glob('*셀러오션*.csv')

In [13]:
dfs = []
for file in sell_list:
    df = pd.read_csv(file)
    print(file)
    dfs.append(df) 

[KB]셀러오션_연수_전처리_final.csv
[KB]셀러오션_채널통합.csv
[KB]셀러오션_채란_전처리_final.csv
[KB]셀러오션_혜진_전처리_final.csv
셀러오션_라벨링.csv


In [14]:
df_sell = pd.concat(dfs, ignore_index = True)
print(df_sell.shape)
df_sell.head()

(343037, 5)


Unnamed: 0                                             review  코더A  코더B  \
0           0  주변사람들도 다 소비를 줄이고 투자를 늘리는 분위기네요 전국에 귀신들린듯 어떻게 이...  NaN  NaN   
1           1  육류(국내산소고기,한우)전문 판매 법인회사 양도 매매 게시글 작성시 최소한의 정보는...  NaN  NaN   
2           2  판매수익 정산 및 사업자 코드 궁금합니다 궁금한 점이 있어서 글써봐요! 1. 현재 ...  NaN  NaN   
3           3  추석 임시공휴일 싫은데… 직장인들은 너무 바라겠지만 온라인은 망… 추석전에 택배 마...  NaN  NaN   
4           4  운수업 창업이나 인수 계획 있으신분 운영권 매각 합니다 자동차부품지입차량6대 직영 ...  NaN  NaN   

   코더C  
0  NaN  
1  NaN  
2  NaN  
3  NaN  
4  NaN

In [15]:
df_sell.drop(columns=['Unnamed: 0'], inplace=True)

In [16]:
df_sell = df_sell.drop_duplicates()
print(df_sell.shape)
df_sell.head()

(134690, 4)


review  코더A  코더B  코더C
0  주변사람들도 다 소비를 줄이고 투자를 늘리는 분위기네요 전국에 귀신들린듯 어떻게 이...  NaN  NaN  NaN
1  육류(국내산소고기,한우)전문 판매 법인회사 양도 매매 게시글 작성시 최소한의 정보는...  NaN  NaN  NaN
2  판매수익 정산 및 사업자 코드 궁금합니다 궁금한 점이 있어서 글써봐요! 1. 현재 ...  NaN  NaN  NaN
3  추석 임시공휴일 싫은데… 직장인들은 너무 바라겠지만 온라인은 망… 추석전에 택배 마...  NaN  NaN  NaN
4  운수업 창업이나 인수 계획 있으신분 운영권 매각 합니다 자동차부품지입차량6대 직영 ...  NaN  NaN  NaN

### 지식인 파일 통합

In [17]:
df_acknow = pd.read_csv('[KB]지식인_유빈새연다운_전처리_final.csv')
print(df_acknow.shape)
df_acknow.head()

(289943, 1)


review
0  올해 월 알제리 여행갈려고 합니다 여자친구한데 도움을 받고 알제 위주로 다닐텐데 현...
1  일본에서 결혼 할 수 있을까요 feat 일본인 여자친구 안녕하세요 세 토종 한국인 ...
2  헤어진 여자친구를 다시 잡고싶어요 안녕하세요 올해로 된 남자입니다 여자친구는 올해로...
3  결혼 계획을 세우고 있는데 여자친구가 아버지 사업으로인한 신용불량자 상태 입니다 혼...
4  저도 여자친구 사귈수있나요 안녕하세요 년생 살 모태솔로입니다 제가 살면서 친구 여자...

In [18]:
df_acknow = df_acknow.drop_duplicates()
print(df_acknow.shape)
df_acknow.head()

(289107, 1)


review
0  올해 월 알제리 여행갈려고 합니다 여자친구한데 도움을 받고 알제 위주로 다닐텐데 현...
1  일본에서 결혼 할 수 있을까요 feat 일본인 여자친구 안녕하세요 세 토종 한국인 ...
2  헤어진 여자친구를 다시 잡고싶어요 안녕하세요 올해로 된 남자입니다 여자친구는 올해로...
3  결혼 계획을 세우고 있는데 여자친구가 아버지 사업으로인한 신용불량자 상태 입니다 혼...
4  저도 여자친구 사귈수있나요 안녕하세요 년생 살 모태솔로입니다 제가 살면서 친구 여자...

In [19]:
len(df_brunch) + len(df_asa) + len(df_sell) + len(df_acknow)

512813

In [20]:
512807 * 0.02

10256.14

### 카테고리별 2%씩 랜덤 샘플 추출

- 라벨링용 채널별 통합 및 1차 전처리 완료 후 데이터 수: 512,807개
- 1차 전처리 후 데이터의 약 3% 라벨링 진행 (카테고리별 2%씩 랜덤 샘플 추출 진행)  
- 2% = approx. 10,256 (*한 사람당 3,419개 내외)


In [21]:
df_brunch_sample = df_brunch.sample(n=None, frac=0.02, replace=False, weights=None, random_state=None, axis=None, ignore_index=False)
df_asa_sample = df_asa.sample(n=None, frac=0.02, replace=False, weights=None, random_state=None, axis=None, ignore_index=False)
df_sell_sample = df_sell.sample(n=None, frac=0.02, replace=False, weights=None, random_state=None, axis=None, ignore_index=False)
df_acknow_sample = df_acknow.sample(n=None, frac=0.02, replace=False, weights=None, random_state=None, axis=None, ignore_index=False)


In [22]:
df_brunch_sample

review  코더A
2060   시효 이야기   시효란 일정한 사실 상태가 일정 기간 계속된 경우  그 사실 상태가...  NaN
4404        중국판 틱톡 인기 숏클립 분석                 데일리 중국 뉴...  NaN
9510   에어팟프로 실리콘 케이스    개 팔아보기  차별점이 없는 제품도 분명 차익거래 a...  NaN
6182   세상 그렇게 사는 거 아닙니다  장사를 하다 보면 블랙 컨슈머  이른바 진상 고객을...  NaN
11964  나를 설레게 하는 도입부   믿고 보는    배우  믿고 듣는    가수  믿고 읽...  NaN
...                                                  ...  ...
1254   블랭크 출신    마케팅 사이드잡 후기  DIO에서는 마케팅도 사이드잡으로 할 수 ...  NaN
5753   AI   화 노동 없는 생산의 시대     이번 chapter부터 다룰 내용은 디플...  NaN
8885   나도 나의 꿈이 공무원이면 좋겠다   마흔 넘은 주부인 내게 음악이 필요한 이유는 ...  NaN
8991       학년도 수능 기행  그냥 지나칠까 하다가 아이들 어릴 때부터 조금씩 남긴 흔...  NaN
10793  플래폼 쓰려면 쓰레기도 같이 보라구요   유튜브 시청자들 사이에는  극혐 하는 광고...  NaN

[242 rows x 2 columns]

In [23]:
df_asa_sample

review  코더A  코더B  코더C
35149  다이소에서 사면 후회하는것들... 개인적으로 다이소 많이 이용하는데 후회하는 물건이...  NaN  NaN  NaN
75765  과일 장사는 어떨까요?? 창업하려 하는데.. 프렌차이즈 과일 장사는 어떨까요?? 많...  NaN  NaN  NaN
41061  매출감소만 확인되면 100만원지급? 어제 퇴근하고 뉴스를 보는데 방역지원금 100만...  NaN  NaN  NaN
19677  미라클 모닝d+75 안녕 나의 하루~ #모닝뮤직🎶 지누션"A-YO" "열받지 가끔씩...  NaN  NaN  NaN
60678  휴대폰창업 휴대폰매장 창업하려고 하는데 다들 어떻게 생각하시나요 전 기본적으로 비추...  NaN  NaN  NaN
...                                                  ...  ...  ...  ...
69499  편의점 구합니다 일매150이상 구합니다 화성 평택 오산 안성 근처 이신분 쪽지 주세...  NaN  NaN  NaN
33669  부동산에 가게를 내놓을려고 할때 부동산에 양도양수나 권리금 걸어놓고 임대를 내놓으려...  NaN  NaN  NaN
35092  하루 일하고 안나온 직원 임금 계산 안나오고 잠수타더니 돈보내라네요. 최저임금으로 ...  NaN  NaN  NaN
37516  희망회복자금,, 좀 봐주세요 저는 4월중순에 분식집을 오픈했어요. 저희지역은 코로나...  NaN  NaN  NaN
11397  정신건강을 위해..일찍 문닫고 집에 갑니다 ㅎ 오픈 한지 2년 조금 넘었는데.. 오...  NaN  NaN  NaN

[1538 rows x 4 columns]

In [24]:
df_sell_sample

review  코더A  코더B  코더C
40090   스마트스토어 여성의류 : ) 스마트스토어 여성의류 이제 3개월차 되어가고 있습니다....  NaN  NaN  NaN
200127         스토어팜 정보 검색으로 가입하게 되었습니다 잘 반갑습니다 반갑습니다 반가워요  NaN  NaN  NaN
26875   오픈마켓별 수수료 정리 공유 드려요~ 아시는 것처럼, 쇼핑몰/카테고리 별로 수수료가...  NaN  NaN  NaN
169612  택배사 cj랑 롯데중에 뭐가 더 배송이 빠를까요 이제 막 시작하는 초보셀러 입니다 ...  NaN  NaN  NaN
63912   산지배송)맛있는 제주도 서귀포 귤 도매 (위탁판매가능 거래전에 먼저 연락처등 검색 ...  NaN  NaN  NaN
...                                                   ...  ...  ...  ...
162672  타오바오 결제 카드등록이 안되요 도와주세요 타오바오 카드결제 할랬더니 문제가 계속 ...  NaN  NaN  NaN
193849  스마트스토어 창업 준비중입니다 잘 반가워요 반갑습니다 안녕하세여 반갑습니다 반가워용...  NaN  NaN  NaN
155608  이랬을때 생기는 문제 뭐가 있을까요 제가 쿠팡 정지고 해서 어차피 다른사업자 있는 ...  NaN  NaN  NaN
3033    쿠팡의 갑질(정시배송+24시간내응대+상담원의양심) 판매일 한지는 10여년되는거같네요...  NaN  NaN  NaN
24469   쿠팡 개떡같은 시스템 때문에 제휴 입고도 못시키네요 아니 쿠팡 시스템은 왜이리 말썽...  NaN  NaN  NaN

[2694 rows x 4 columns]

In [25]:
df_acknow_sample1 = df_acknow_sample[:1638]

In [26]:
df_acknow_sample2 = df_acknow_sample[1638:2363]

In [27]:
df_acknow_sample3 = df_acknow_sample[2363:]

### CSV 파일로 저장

In [28]:
df_brunch_sample.to_csv('브런치_라벨링.csv', encoding = 'utf-8-sig')
df_asa_sample.to_csv('아싸_라벨링.csv', encoding = 'utf-8-sig')
df_sell_sample.to_csv('셀러오션_라벨링.csv', encoding = 'utf-8-sig')
df_acknow_sample1.to_csv('지식인1_라벨링.csv', encoding = 'utf-8-sig')
df_acknow_sample2.to_csv('지식인2_라벨링.csv', encoding = 'utf-8-sig')
df_acknow_sample3.to_csv('지식인3_라벨링.csv', encoding = 'utf-8-sig')


In [29]:
df_brunch.to_csv('[KB]브런치_채널통합.csv', encoding='utf-8-sig')
df_asa.to_csv('[KB]아싸_채널통합.csv', encoding='utf-8-sig')
df_sell.to_csv('[KB]셀러오션_채널통합.csv', encoding='utf-8-sig')
df_acknow.to_csv('[KB]지식인_채널통합.csv', encoding='utf-8-sig')

## Test 데이터파일 생성

In [30]:
bs = df_brunch_sample.index

In [31]:
df_acknow

review
0       올해 월 알제리 여행갈려고 합니다 여자친구한데 도움을 받고 알제 위주로 다닐텐데 현...
1       일본에서 결혼 할 수 있을까요 feat 일본인 여자친구 안녕하세요 세 토종 한국인 ...
2       헤어진 여자친구를 다시 잡고싶어요 안녕하세요 올해로 된 남자입니다 여자친구는 올해로...
3       결혼 계획을 세우고 있는데 여자친구가 아버지 사업으로인한 신용불량자 상태 입니다 혼...
4       저도 여자친구 사귈수있나요 안녕하세요 년생 살 모태솔로입니다 제가 살면서 친구 여자...
...                                                   ...
289938                 아산 문의 저번주에 뭐떼야 하는데 아산 구경찰서 어디 인가요 
289939                             내공 빈티지 옷인데 정보 좀 알려주세요 
289940   재래시장가서 쑥가루랑 흑임자가루랑 콩가루사려면 떡집찾아가면된나요 아니면 다른 빼는...
289941   년대 빈티지 컵 포도송이 펀칭글라스 브랜드 이름을 알 수 있을까요 쿠팡같은 광고는...
289942  빈티지샵 폴로 둘다 가품일까요 정품이라고 써두긴 했는데 좀 찝찝해서요 폴로 홈페이지...

[289107 rows x 1 columns]

In [32]:
df_brunch_test = df_brunch.drop(df_brunch_sample.index)
df_asa_test = df_asa.drop(df_asa_sample.index)
df_sell_test = df_sell.drop(df_sell_sample.index)
df_acknow_test = df_acknow.drop(df_acknow_sample.index)

In [33]:
test_all = pd.concat([df_brunch_test, df_asa_test, df_sell_test, df_acknow_test], axis=0)

In [34]:
test_all.to_csv('test_all.csv', encoding='utf-8-sig')

## 추가 전처리

In [35]:
test_all

review  코더A  코더B  코더C
0          프리패스  상황에 맞는 OOTD가 이렇게나 중요하다 택배박스도 보여주고 어디 ...  NaN  NaN  NaN
1       틈은 흠일까  지난 시간들을 돌이켜 보면 나는 원래 틈이 많은 사람이었다  손해를 ...  NaN  NaN  NaN
2                                            무인모텔     계속   NaN  NaN  NaN
3       무인문구점  그게 되겠어      아이가 생기고 나서는 주말은 온전히 아이를 위한 ...  NaN  NaN  NaN
4                           무인키즈카페 창업일기  편  상가 인테리어 계약     NaN  NaN  NaN
...                                                   ...  ...  ...  ...
289938                 아산 문의 저번주에 뭐떼야 하는데 아산 구경찰서 어디 인가요   NaN  NaN  NaN
289939                             내공 빈티지 옷인데 정보 좀 알려주세요   NaN  NaN  NaN
289940   재래시장가서 쑥가루랑 흑임자가루랑 콩가루사려면 떡집찾아가면된나요 아니면 다른 빼는...  NaN  NaN  NaN
289941   년대 빈티지 컵 포도송이 펀칭글라스 브랜드 이름을 알 수 있을까요 쿠팡같은 광고는...  NaN  NaN  NaN
289942  빈티지샵 폴로 둘다 가품일까요 정품이라고 써두긴 했는데 좀 찝찝해서요 폴로 홈페이지...  NaN  NaN  NaN

[502557 rows x 4 columns]

In [ ]:
test_all.review